### ISSUE TITLE
Serializing Date field in django-rest-marshmallow results in an error

### ISSUE THREAD

**Issue by spokzers** [marshmallow-code/django-rest-marshmallow#110]
I am using `django-rest-marshmallow` along with `django-rest-framework` to build simple APIs.
The following works if I dont include the `created_at` field in the serializer and gets the results as expected but upon inclusion, it throws an exception.

`models.py`

    from django.db import models

    class Category(models.Model):
        name = models.CharField(max_length=50)
        created_at = models.DateField()

`serializers.py`

    from rest_marshmallow import Schema, fields

    class CategorySerializer(Schema):
        id = fields.Integer()
        name = fields.String()
        created_at = fields.Date() # works, if I dont include this

`views.py`

    from rest_framework import viewsets

    class CategoryViewSet(viewsets.ModelViewSet):
        queryset = Category.objects.all()
        serializer_class = CategorySerializer

This results in the following error
```
Django version 2.2.5, using settings 'app.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
Internal Server Error: /api/v1/categories/
Traceback (most recent call last):
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/rest_framework/viewsets.py", line 114, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/rest_framework/views.py", line 505, in dispatch
    response = self.handle_exception(exc)
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/rest_framework/views.py", line 465, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/rest_framework/views.py", line 476, in raise_uncaught_exception
    raise exc
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/rest_framework/views.py", line 502, in dispatch
    response = handler(request, *args, **kwargs)
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/rest_framework/mixins.py", line 45, in list
    serializer = self.get_serializer(queryset, many=True)
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/rest_framework/generics.py", line 110, in get_serializer
    return serializer_class(*args, **kwargs)
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/rest_marshmallow/__init__.py", line 36, in __init__
    MarshmallowSchema.__init__(self, **schema_kwargs)
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/marshmallow/schema.py", line 384, in __init__
    self._init_fields()
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/marshmallow/schema.py", line 915, in _init_fields
    self._bind_field(field_name, field_obj)
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/marshmallow/schema.py", line 977, in _bind_field
    field_obj._bind_to_schema(field_name, self)
  File "/home/abcdefg/mpk/lib/python3.7/site-packages/marshmallow/fields.py", line 1181, in _bind_to_schema
    or getattr(self.root.opts, self.SCHEMA_OPTS_VAR_NAME)
AttributeError: 'NoneType' object has no attribute 'opts'
```

**Comment by sloria** [comment]
What version of marshallow and Django-rest-marshmallow are you using?

**Comment by spokzers** [comment]
Using these
```
django-rest-marshmallow==4.0.1
marshmallow==3.1.1
```

**Comment by sloria** [comment]
OK. Unfortunately I won't have time to look into this today. I would gladly review and accept a PR for this.

**Comment by vzts** [comment]
The above calls seem to be executed while deciding format.
My temporary workaround was to explicitly specify format like `DateTime(format='iso')`.
Guess this is related to https://github.com/marshmallow-code/marshmallow/issues/1357#issuecomment-523471107

**Comment by sloria** [comment]
fixed in 4.0.2
